mautic/mautic
# Set all fields as listable

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | N
| Automated tests included? | N
| Related user documentation PR URL | /
| Related developer documentation PR URL | /
| Issues addressed (#s or URLs) | /
| BC breaks? | N
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
https://github.com/mautic/mautic/pull/5906 fixed the conditions for segment filter fields. Although it fixes the issue successfully the problem is that some default custom fields:
```
title
firstname
lastname
company
position
email
points
city
state
zipcode
country
companyemail
companystate
companycountry
companyname
companynumber_of_employees
companyindustry
companydescription
```
Are set as not listable. `is_listable` is a flag to show the field in the segment filter. Since the condition was not working it was showing all fields and all was good. When that condition was fixed some important fields disappeared.

So this PR sets all fields as listable in a migration for active instances and for new installations.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Try to create a segment filter based on the email field on current staging - cannot be done since the email field is not in the list.

#### Steps to test this PR:
0. Run `app:console doctrine:migrations:migrate`
1. Try to create a segment filter based on the email field - easy peasy. The email field is there.
2. Create new installation with this PR. Notice email field is available in the segment filter select box.
